### DISCUSSION TITLE
Balena support for ot-br-posix

### DISCUSSION THREAD

**Discussion by beriberikix** [openthread/openthread#7565] [Q&A] [answered]
I'm interesting in enabling a few testers with a managed Raspberry Pi-based Border Router. [Balena](https://www.balena.io/) is a container-based device management that seems to fit the bill (and has a generous free tier if others want to do the same.)

I slightly [modified](https://github.com/beriberikix/ot-br-posix/blob/main/etc/balena/Dockerfile.template) the original Dockerfile to be compatible with Balena's requirements but I'm [getting an error](https://gist.github.com/beriberikix/3cea5819f793b02d4a7afaa4e9aff327) with little information. It's hard to make heads or tails with that much output, so I'm wondering if anyone can spot the issue?

**Comment by jwhui** [comment, marked as answer]
It appears that the `git submodule` command is failing because the working directory is not a git repo. As a result, OpenThread fails to build.

https://gist.github.com/beriberikix/3cea5819f793b02d4a7afaa4e9aff327#file-ot-br-balena-errors-L320

**Comment by mpous** [comment]
@beriberikix do you have the latest Dockerfile that you tested? i could deploy properly the Dockerfile on this repo but not sure it works!

**Reply by beriberikix** [reply]
Yup! Only slightly modified from the original but TBH my experience with Balena is limited so I kinda didn't know what I was doing :) https://github.com/beriberikix/ot-br-posix/blob/main/etc/balena/Dockerfile.template

**Reply by beriberikix** [reply]
Here's more documentation on the Docker support: https://openthread.io/guides/border-router/docker
